### PR TITLE
Feature/sdev 3122 match all hgvs tests

### PR DESCRIPTION
**Pull request by dustinbleile**
Increase matching tests for small mutations.

New Features:
 * GERO-299 - annotation - remove conflicting category variants of both nonsense and missense, simultaneously.

Improvements:
 * Update TERT promoter small mutation test TERT:c.1-124C>T
 * Assume default variant types

Refactor:
 * removed nonstandard 'verbose' logging level (only used in one line)

### CODE REVIEW

File: ipr/annotate.py [hunk 1]
```diff
@@ -369,23 +370,38 @@ def annotate_positional_variants(
...
-            variant = row.get(var_key, '')
+            variant = row.get(var_key)
             if not variant:
                 continue
             try:
                 matches = gkb_match.match_positional_variant(graphkb_conn, variant)
 
+                # GERO-299 - check for conflicting nonsense and missense categories
+                missense = [m for m in matches if 'missense' in m.get('displayName', '')]
+                nonsense = [m for m in matches if 'nonsense' in m.get('displayName', '')]
```
Comment (mathieulemieux) [inline ipr/annotate.py:387]: Humm, the type is not always in the displayName; e.g. 'TP53:p.M237I' has a type 'missense mutation'.
Is **m.get('type', {}).get('displayName', '')** would be preferable ?

File: ipr/annotate.py [hunk 2]
```diff
@@ -369,23 +370,38 @@ def annotate_positional_variants(
...
+                if missense and nonsense:
+                    conflict_names = sorted(
+                        set([m.get('displayName', '') for m in nonsense + missense])
+                    )
+                    logger.error(
+                        f'Conflicting nonsense and misense categories for {variant}: {conflict_names}'
+                    )
+                    logger.error("GERO-299 Dropping all conflicting missense/nonsense categories.")
+                    matches = [m for m in matches if m not in missense and m not in nonsense]
```
Comment (mathieulemieux) [inline ipr/annotate.py:396]: So we are dropping those match altogether if conflicting. Shouldn't we still keep the 'missense' one?

File: ipr/annotate.py [hunk 3]
```diff
@@ -159,6 +159,7 @@ def get_statements_from_variants(
         + [f'evidence.{p}' for p in GENERIC_RETURN_PROPERTIES]
         + [f'relevance.{p}' for p in GENERIC_RETURN_PROPERTIES]
         + [f'evidenceLevel.{p}' for p in GENERIC_RETURN_PROPERTIES]
+        # TODO: GERO-289 - IPR evidence equivalents
```
Comment (mathieulemieux) [inline ipr/annotate.py:162]: Ok. I added a proposition for that here: https://github.com/bcgsc/pori_ipr_python/commits/feature/GERO-289-get-evidencelevel-mapping

File: ipr/annotate.py (hunk 1 above)
Comment (dustinbleile) [reply ipr/annotate.py]: It was only the CategoryVariants that had the issue.

But I've tried altering this to get all the variant types and then trying to resolve the conflicts.

File: ipr/annotate.py (hunk 2 above)
Comment (dustinbleile) [reply ipr/annotate.py]: Tried adding some logic to keep the 'missense' the appropriate category, if the additional matches can be used to determine which category is correct.

File: ipr/annotate.py (hunk 3 above)
Comment (dustinbleile) [reply ipr/annotate.py:162]: Yes those changes are helpful.
However, I will make a separate pull request for those.

File: ipr/annotate.py (hunk 2 above)
Comment (mathieulemieux) [reply ipr/annotate.py]: ok, I tested it and it seems to work. Ideally

**Comment by mathieulemieux** [comment]
Can we fix the unsuccessful build check?

**Review by mathieulemieux — APPROVED** [review]
Approved despite unsuccessful build check, which will need to be addressed I guess

**Comment by mattdoug604** [comment]
> Approved despite unsuccessful build check, which will need to be addressed I guess

Re-running the build worked this time. The error was that GraphKB was refusing connections - I guess too many builds tried to connect at once?

**Review by mattdoug604 — APPROVED** [review]

**Pull request merged**